buttonwall/modes/__init__.py

```diff
@@ -18,37 +29,84 @@ def start(self):
+            reaction_time = button.reaction_time_end - button.reaction_time_start
+            self.player_one.reaction_time.append(reaction_time)
+        elif button.id % 2 == 1 and not button.color.compare(ColorWhite()):
+            self.player_two.score = self.player_two.score +1
+            reaction_time = button.reaction_time_end - button.reaction_time_start
+            self.player_two.reaction_time.append(reaction_time)
+
+
+        '''If there is pressed button of wrong color'''
+        # if button.id % 2 == 0 and not button.color.compare(ColorGreen()):
+        #     self.player_one.score = self.player_one.score -1
+        # elif button.id % 2 == 1 and not button.color.compare(ColorGreen()):
+        #     self.player_two.score = self.player_two.score -1
+
         self.pressed_event.set()
 
-    def button_releasse_callback(self, future: asyncio.Future):
+
+    def button_releasse_callback(self, future: asyncio.Future, button: Button):
         ''' If there is released some button '''
 
-        future.result().add_done_callback(self.button_releasse_callback)
+        future.result().add_done_callback(partial (self.button_releasse_callback, button= button))
         self.pressed_event.clear()
 
+    def timeout(self):
+        self.pressed_event.set()
+
     async def run(self):
         '''
         Simple method to demonstrate functionality
         If there is some button pressed then starts random color changing cycle
         '''
-        colors = [ColorWhite(), ColorRed(), ColorGreen(), ColorBlue(), ColorPurple()]
+       
+        colors = [ColorRed(), ColorGreen(), ColorBlue(), ColorPurple()]
         if not self.running:
             self.running = True
 
             for button in self.manager.buttons.values():
-                button.pressed_future.add_done_callback(self.button_pressed_callback)
-                button.released_future.add_done_callback(self.button_releasse_callback)
+                button.pressed_future.add_done_callback(partial(self.button_pressed_callback, button=button))
+                button.released_future.add_done_callback(partial(self.button_releasse_callback, button=button))
+            
+            timeout = 60
+            loop = asyncio.get_event_loop()
+            timeoutTimerHandle = loop.call_later(timeout, self.timeout)
 
             while self.running:
+
                 await self.pressed_event.wait()
+                
+                timeoutTimerHandle.cancel()
+
+                while True: 
 
-                for button in self.manager.buttons.values():
                     if not self.pressed_event.is_set():
                         break
 
-                    button.set_color(choice(colors))
-                    await asyncio.sleep(0.1)
+                    for button in self.manager.buttons.values():
+                        button.set_color(ColorWhite())
+
+                    button = choice(self.manager.buttons)
```

Comment:
Občas sa vysypáva, neviem prísť na dôvod prečo, niekedy sa to vysype hneď, niekedy to v pohode beží, niekedy to ide napríklad 5 cyklov a potom to padne. Neprišla som však na dôvod. Nižšie je odkaz na vypisovanú hlášku:
https://gist.github.com/PitoRito/416ada50012fbb4df9efe9ed8cde313c